providers/summary/scancode.js

```diff
@@ -84,10 +91,26 @@ class ScanCodeSummarizer {
     if (!licenses) return null
     const list = setToArray(licenses)
     if (!list) return null
-    return list
-      .map(normalizeSpdx)
-      .filter(x => x)
-      .join(' AND ')
+    if (list.length === 1) return list[0]
+    return list.map(x => `(${x})`).join(' AND ')
+  }
+
+  _createExpression(rule, licenseKey) {
+    const license_expression = get(rule, 'license_expression')
+    if (!license_expression) return normalizeSpdx(licenseKey)
+    let result = license_expression
+    rule.licenses.forEach(licence => {
+      result = result.replace(licence, normalizeSpdx(licence) || '').trim()
+    })
+    new Array('AND', 'OR', 'WITH').forEach(operator => {
```

Comment:
do we ever get license expressions from ScanCode that have leading or trailing AND, OR or WITH? Seems like that would be a bug in ScanCode?

BTW, matching for leading/trailing OR seems is a bit aggressive .

Reply:
The thinking here was that if `normalizeSpdx()` removes an identifier then we will have an invalid expression

`mit OR junk` => `mit OR` => `mit`

Reply:
^^^ see the conversation from this morning about not filtering out non-spdx licenses and leaving a marker that shows there is something non-spdx here